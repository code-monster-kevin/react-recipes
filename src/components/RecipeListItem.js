import React from 'react';
import PropTypes from 'prop-types';

const RecipeListItem = ({ recipe, favourited, onClick, onFavourited }) => (
  <li
    key={recipe.id}
    className="py2 border-bottom border-bottom-dashed pointer"
    onClick={() => onClick(recipe.id)}
  >
    <span
      className="mr1"
      role="img"
      aria-label="Favourite"
      onClick={e => {
        e.stopPropagation();
        onFavourited(recipe.id);
      }}
    >
      {favourited ? '‚ù§Ô∏è' : 'üñ§'}
    </span>
    <span>{recipe.name}</span>
    <span>{recipe.category}</span>
  </li>
);

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  favourited: PropTypes.bool,
  onClick: PropTypes.func,
  onFavourited: PropTypes.func,
};

export default RecipeListItem;
