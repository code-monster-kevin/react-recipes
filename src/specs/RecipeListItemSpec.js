import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../components/RecipeListItem';

const mockRecipe = {
  id: 1,
  name: 'Mock Recipe',
  category: 'Mock Category',
};

describe('<RecipeListItem />', () => {
  test('No recipe props input', () => {
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Valid recipe props input', () => {
    const component = renderer.create(<RecipeListItem recipe={mockRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Render favourite state', () => {
    const component = renderer.create(
      <RecipeListItem recipe={mockRecipe} favourited={true} />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
