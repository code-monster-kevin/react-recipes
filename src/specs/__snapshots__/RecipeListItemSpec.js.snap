// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> No recipe props input 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="Favourite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    üñ§
  </span>
  <span />
  <span />
</li>
`;

exports[`<RecipeListItem /> Render favourite state 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="Favourite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    ‚ù§Ô∏è
  </span>
  <span>
    Mock Recipe
  </span>
  <span>
    Mock Category
  </span>
</li>
`;

exports[`<RecipeListItem /> Valid recipe props input 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="Favourite"
    className="mr1"
    onClick={[Function]}
    role="img"
  >
    üñ§
  </span>
  <span>
    Mock Recipe
  </span>
  <span>
    Mock Category
  </span>
</li>
`;
